code-styling/mysql/README.md

```diff
@@ -0,0 +1,574 @@
+---
+permalink: /code-styling/mysql/
+---
+
+# Dealerdirect style guide: MySQL
+
+This guide defines the MySQL coding standards we use at Dealerdirect. The intent
+of this guide is to help developers create MySQL queries that are readable,
+portable, easy to maintain and bug-free.
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC 2119][rfc2119].
+
+## Table of Content
+
+- [Files](#files)
+
+- [Inline queries](#inline-queries)
+
+  - [Queries in PHP files](#queries-in-php-files)
+
+- [Database design](#database-design)
+
+  - [Tables](#tables)
+  - [Columns](#columns)
+  - [Foreign keys](#foreign-keys)
+
+- [Formatting](#formatting)
+
+  - [Select queries](#select-queries)
+  - [Joins](#joins)
+  - [Delete queries](#delete-queries)
+  - [Insert queries](#insert-queries)
+  - [Update queries](#update-queries)
+  - [Other select criteria](#other-select-criteria)
+  - [Grouping](#grouping)
+  - [Ordering](#ordering)
+  - [Limit](#limit)
+
+<!-- ======================================================================= -->
+
+## Files
+
+This section only applies to files that only contain MySQL statements. Files
+that contain queries as part of programming code in another language SHOULD
+adhere to the naming conventions for that language.
+
+- All MySQL filenames MUST have the `.sql` extension.
+
+- File names MAY NOT contain characters other than letters (a to z), numbers (0
+  to 9), hyphen, `-` and dot `.`.
+
+- File names MUST NOT start or end with a hyphen or dot.
+
+- File names MUST NOT have consecutive hyphens or dots.
+
+## Inline queries
+
+Inline queries are sql queries in a file of another programming language, such
+as a PHP file with sql queries.
+
+### Queries in php files
```

Comment:
I am unsure whether this belongs here or iw we should add/update this to a PHP standard..